pkgs/development/libraries/openenclave/default.nix

```diff
@@ -0,0 +1,113 @@
+{ type }:
+assert builtins.elem type [ "sgx" ];
```

Comment:
What is this?

Reply:
This is a selector for the target we want to build OpenEnclave for. It can be built to target either Intel SGX or ARM TrustZone. In the future I want to add TrustZone support, however this currently only supports SGX, so I added an assertion to ensure that it is set to SGX.